app/views/handlers/question.py

```diff
@@ -135,25 +141,28 @@ def _get_answers_for_question(self, question_json) -> dict[str, Any]:
 
         return answer_value_by_answer_id
 
-    def _get_list_add_question_url(self, params):
+    def _get_list_add_question_url(self, params: dict) -> str | None:
         block_id = params["block_id"]
         list_name = params["list_name"]
         list_items = self._questionnaire_store.list_store[list_name].items
         section_id = self._schema.get_section_id_for_block_id(block_id)
 
         if self._is_list_just_primary(list_items, list_name) or not list_items:
             return Location(
-                section_id=section_id, block_id=block_id, list_name=list_name
+                # type ignore: section_id is always valid when this is called
+                section_id=section_id,
+                block_id=block_id,
+                list_name=list_name,  # type: ignore
             ).url(previous=self.current_location.block_id)
 
-    def _is_list_just_primary(self, list_items, list_name):
+    def _is_list_just_primary(self, list_items: list[str], list_name: str) -> bool:
         return (
             len(list_items) == 1
             and list_items[0]
             == self._questionnaire_store.list_store[list_name].primary_person
         )
 
-    def _get_answer_action(self):
+    def _get_answer_action(self) -> dict[str, str] | None:
```

Comment:
action can also have a property called "params" which is a dictionary so it would be `dict[str, str | dict]` which isn't really worth writing so you can change it to just `dict | None`

Reply:
still outstanding